exchange/src/curl/curl.cpp

```diff
@@ -15,8 +17,55 @@ write_callback(void* contents, size_t size, size_t nmemb, void* userp)
     return size * nmemb;
 }
 
-glz::json_t
-firebase_request(
+static size_t
+file_write_callback(void* contents, size_t size, size_t nmemb, FILE* stream)
+{
+    return fwrite(contents, size, nmemb, stream);
+}
+
+void
+request_to_file(
+    const std::string& method, const std::string& url, const std::string& filepath,
+    const std::string& data
+)
+{
+    CURL* curl;
+    CURLcode res;
+    FILE* fp;
+
+    curl = curl_easy_init();
+    if (curl) {
+        fp = fopen(filepath.c_str(), "wb");
+        curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
+        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, file_write_callback);
```

Comment:
```suggestion
        curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, fwrite);
```

No need to wrap `fwrite` if you just pass it.